cast/java/ecj/src/main/java/com/ibm/wala/cast/java/translator/jdt/JDTJava2CAstTranslator.java

```diff
@@ -929,6 +943,127 @@ else if ((n.getModifiers() & Modifier.ABSTRACT) != 0) {
     return new ProcedureEntity(mdast, n, classBinding, memberEntities, context, annotations);
   }
 
+  public CAstNode visit(LambdaExpression n, WalkContext oldContext) {
+
+    ITypeBinding typeBinding = n.resolveTypeBinding().getErasure();
+    List<?> parameters = n.parameters();
+    List<CAstType> castTypes = new ArrayList<>();
+    List<String> castNames = new ArrayList<>();
+
+    for (Object obj : parameters) {
+      VariableDeclarationFragment tmp = (VariableDeclarationFragment) obj;
+      CAstType td = fTypeDict.getCAstTypeFor(tmp.resolveBinding().getType());
+      castTypes.add(td);
+      String tmpName = tmp.getName().getIdentifier();
+      castNames.add(tmpName);
+    }
+
+    // pass in memberEntities to the context, later visit(New) etc. may add classes
+    final Map<CAstNode, CAstEntity> memberEntities = HashMapFactory.make();
+    final MethodContext context =
+        new MethodContext(oldContext, memberEntities); // LEFTOUT: in polyglot there is a
+    // class context in between method and
+    // root
+
+    CAstNode mdast = fFactory.makeNode(CAstNode.RETURN, visitNode(n.getBody(), context));
```

Comment:
Do we want this to be a `RETURN` node even for lambdas that don't return a value (like a `Consumer`)?

Reply:
it seems harmless, but i got rid of it.

Reply:
added a test with a void lambda.